### ISSUE TITLE
Weird SRT321 behaviour and possible Secure app for Homey Pro

### ISSUE THREAD

**Issue by oleroes** [hankolsen/com.securemeters.horstmann#42]
Hi!

This comment is from my thread on devices being switched on when they shouldn't be:

I have discovered another problem that I can’t explain that seems to be switching some devices on as well, and that is that two of my Secure SRT321 thermostats (ID 157 and 161) have sent “Switch Binary Set (ON)” to all my devices it looks like. (See image.) And there is something wrong in the Secure app for Homey, but since I have several other SRT321’s that haven’t done this (at least not yet) i suspect its only in the description of the association groups. I will try and share this thread with the developer of the app.

But what do you guys think, are the thermostats broken or will an exclude and include most likely fix it?

Or could it be a bug in the Secure app for Homey? There is a screen grab of the Zniffer file showing the issue.

Link to thread: https://community.homey.app/t/devices-switching-on-when-they-should-be-off/77047?u=ole_rosset

**Comment by ViaraiX** [comment]
It sounds like you’ve associated it with other devices. I would check the details of all devices being activated and yes, removing and restarting would possibly work, but wouldn’t answer why.

Cheers.
________________________________
From: oleroes ***@***.***>
Sent: Friday, March 3, 2023 1:43:38 PM
To: hankolsen/com.securemeters.horstmann ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [hankolsen/com.securemeters.horstmann] Weird SRT321 behaviour and possible Secure app for Homey Pro (Issue #42)


Hi!

This comment is from my thread on devices being switched on when they shouldn't be:

I have discovered another problem that I can’t explain that seems to be switching some devices on as well, and that is that two of my Secure SRT321 thermostats (ID 157 and 161) have sent “Switch Binary Set (ON)” to all my devices it looks like. (See image.) And there is something wrong in the Secure app for Homey, but since I have several other SRT321’s that haven’t done this (at least not yet) i suspect its only in the description of the association groups. I will try and share this thread with the developer of the app.

But what do you guys think, are the thermostats broken or will an exclude and include most likely fix it?

Or could it be a bug in the Secure app for Homey? There is a screen grab of the Zniffer file showing the issue.

Link to thread: https://community.homey.app/t/devices-switching-on-when-they-should-be-off/77047?u=ole_rosset

**Comment by oleroes** [comment]
> It sounds like you’ve associated it with other devices. I would check the details of all devices being activated and yes, removing and restarting would possibly work, but wouldn’t answer why. Cheers.
> […](#)

I have associated two devices that are heaters it is meant to control, but the thermostat has sent switch binary set to 50+ devices.

**Comment by ViaraiX** [comment]
Delete the switch completely and start again. Sounds like a config issue.

Either way, the secure meter app is working fine, it’ll be something else from the limited info.
________________________________

**Comment by oleroes** [comment]
> Delete the switch completely and start again. Sounds like a config issue. Either way, the secure meter app is working fine, it’ll be something else from the limited info.
> […](#)

Yes, it could be that it is only the association group descriptions that are wrong in the app.

**Comment by ViaraiX** [comment]
Yes, it’s likely you’ve tried to direct associate it on a group designed to send to all. Be careful with that.

I would either use the hub to do the control switching or if the switch is attached to a zwave device, directly associate them that way.

You can’t directly associate a virtual device with the secure controller, as one device isn’t real and likely is initiated from within your zwave hub/controller.

Good luck! Be fun to hear about what caused this? But it’s likely something in the way you’ve tried to associate.
________________________________

**Comment by oleroes** [comment]
Hi and thanks for you input so far. I have attached two images, one showing default associations after including the SRT321 (1 in all groups) and the other image is how I have set them up. To be honest I am hoping I have set it up wrong as that would explain why this is happening. I have one or two (as in this image) devices in group 2 for all my thermostats and they are wall plugs with heaters.
![Skjermbilde 2023-03-05 kl  17 02 40](https://user-images.githubusercontent.com/126885905/222972450-3a7e20f5-2aea-48ef-89b7-acba8525773d.png)
![Skjermbilde 2023-03-05 kl  17 03 16](https://user-images.githubusercontent.com/126885905/222972451-d4189cfe-ad5a-49ba-bd05-d17d83f3331b.png)

**Comment by ViaraiX** [comment]
I would make a scene to drive the wall thermostats based on checking. The SRT is designed to directly associate with a boiler switch .

However, check the manual for the groups and see what the groups actually do for each product, including heaters.

Group 1 is almost always lifeline.. so you should leave those to default.

Attached is the SRT 321 manual on groups and what they do, but depends on your thermostats as well.

Hope that helps.
________________________________

**Comment by oleroes** [comment]
The SRT321 is also meant for controlling electrical heating, depends on how you set the DIL switches. I have the manual and that is why I have set it the way it is. Group 1 is not specified as Lifeline in my manual and I only have 5 groups, but there seems to be a newer version with 6 groups where group 1 is Lifeline. I any case I have put node 1 back in group 1 since it was there by default. I would love a comment from the app developer as well since he must have some idea on how to best set up the associations. I don't see why the controller should be in all groups?
![Skjermbilde 2023-03-06 kl  00 00 37](https://user-images.githubusercontent.com/126885905/222990896-5d6a11e4-bb7c-451a-bdac-cbe4d4214aa3.png)

**Comment by ViaraiX** [comment]
The controller is in all groups because that’s the default. Your device is adopted and the controller can see all groups.

The developer has done an excellent job, I am unsure if you’ll get help relating to specific association to your own devices, as this is outside the realms of the fault of the app, I suspect. However: I’m a happy to be wrong ☺️
________________________________

**Comment by hankolsen** [comment]
Sorry if i can't help out much. All I know is that I have my wall socket plugs associated to Group 3.
The SRT321 is acting as the thermostat and the wall socket plugs control the heaters.

**Comment by oleroes** [comment]
Yeah, then you have the newer version of the SRT321 I guess, if I place the wall plugs in group 3 like the description in the app says, nothing happens when the thermostat calls for heat, but they do switch on and off if placed in group 2. Do you have node 1 in all groups in addition to your wall plugs?

**Comment by hankolsen** [comment]
Yes, node 1 is added to all groups by default

**Comment by oleroes** [comment]
Thanks for your reply Hank, I appreciate all the help I can get. You have the same setup as me, only with newer version of the SRT321 which has the extra Lifeline group, mine only has 5 groups. (Could that cause a problem with the app?) So with the manual for my version of the SRT321, do you think my associations should look like this? Or should my Homey be removed from group 1 and 2 which is meant for controlling other nodes? I have 8 of these thermostats and the fact that 4 of them has sent ON-commands to all of my devices so far makes me think that its not the thermostats that are broken, but that my settings are wrong somehow. Thanks!
![Skjermbilde 2023-03-06 kl  10 43 28](https://user-images.githubusercontent.com/126885905/223076833-7d279961-0681-46bd-a921-6a398d7de6b0.png)

**Comment by hankolsen** [comment]
In my setup I don't have node 1 in the group I'm controlling with the SRT321. Not sure if it makes any difference though.

![IMG_B2A72C21A9B0-1](https://user-images.githubusercontent.com/1008390/223078968-3d892416-922d-4f88-88a2-b3f4ca3e18c8.jpeg)
